src/app/app.module.ts

```diff
@@ -1,18 +1,46 @@
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core'
+import { BrowserModule } from '@angular/platform-browser'
+import { MatIconModule } from '@angular/material/icon'
+import { MatInputModule } from '@angular/material/input'
+import { MatCardModule } from '@angular/material/card'
 
-import { AppRoutingModule } from './app-routing.module';
-import { AppComponent } from './app.component';
+import { AppRoutingModule } from './app-routing.module'
+import { AppComponent } from './app.component'
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
+
+import { HeaderComponent } from './components/header/header.component'
+import { UserLoginComponent } from './components/user-login/user-login.component'
+import { SearchSectionComponent } from './components/search-section/search-section.component'
+import { VideoCourseListComponent } from './components/video-course-list/video-course-list.component'
+import { VideoCourseComponent } from './components/video-course/video-course.component'
+import { FooterComponent } from './components/footer/footer.component'
+import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component'
+import { LogoComponent } from './components/logo/logo.component'
+import { CoursesPageComponent } from './pages/courses-page/courses-page.component'
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    HeaderComponent,
+    UserLoginComponent,
+    SearchSectionComponent,
+    VideoCourseListComponent,
+    VideoCourseComponent,
+    FooterComponent,
+    BreadcrumbsComponent,
+    LogoComponent,
+    CoursesPageComponent,
```

Comment:
You can try to move components into some smaller modules as well. For example, VideoCourseModule for video course related components, SharedModule that can store components that will be reused all over several other modules in application, etc.

Also in the latest Angular version there is a really cool feature of standalone components, that lets you organise code specifically around each component and you can avoid using modules at all. It is up to you what approach to take, but I would recommend to try to create at least one component as a standalone, to understand how it works.

Reply:
Ok, I have it tried video-course-list, please check if correctly and if I should do something additionally.

Reply:
Ok, I have tried to make video-course-list.component as standalone, please check if correctly.